### DISCUSSION TITLE
Dependent queries do not refetchOnMount.

### DISCUSSION THREAD

**Discussion by lisherwin** [TanStack/query#5241] [Q&A] [answered]
Hi React Query folks,

I have a question regarding refetchOnMount and dependent queries. I have a simple react query hook A, and another react query hook B that depends on the data returned from A (B's `enabled` is conditioned on the data of A). On mount, I'd like for A to refetch and for B to refetch right after A. 

However, when I set `refetchOnMount = 'always'` for both A and B, A successfully refetches on mount, but B does not because B's enabled condition is not satisfied on mount because A's data fetch is asynchronous and RQ's in-memory cache is not yet populated. B "misses" the refetch on mount because its condition is not yet satisfied and also does not refetch when the condition is satisfied when the async function returns. 

This is better shown in my sandbox. To see the 2nd query not refetch on mount, you'll need to refresh the page:
https://codesandbox.io/s/beautiful-pare-eyvzj1?file=/src/App.jsx

Is there a way to tell React Query to try the refetchOnMount once B's condition is satisfied? If not, (I suspect there isn't) what's the best way to get B to refetch on mount AND when A returns satisfying data?

**Comment by TkDodo** [comment]
since you are sing `cats` (the value returned from `A`) inside the queryFn of query B, it needs to be part of the QueryKey for that query.

But you are right in the sense that `refetchOnMount: 'always'` doesn't do much in this situation because the event that will trigger a refetch eventually is not the mount, but the key change.

**Reply by lisherwin** [reply]
Thanks for the fast response! I updated the `paws` query to have `cats` in its query key, though as you mentioned this doesn't trigger B's refetch on mount after `A` returns a non-empty list from **A's** refetch on mount.

That being said, is the only way to get B to "refetch on mount" by setting stale timeout to 0? The "refetch" due to key change doesn't happen because the `staleTime` is Infinity.

Happy to hear any thoughts!

**Reply by lisherwin** [reply]
@TkDodo Just updated my comment a bit after some more digging in. It seems like the only way to get this refetch to happen due to key change is by setting `staleTime` to 0 since query `B` won't get the "on mount refetch"? In the sandbox, `B` doesn't get retriggered due to `A` returning `cats` because its `staleTime` is Infinity.

**Reply by TkDodo** [reply, marked as answer]
yes true. For key changes, there is no `refetchOn...` event that you could set to `'always'`, so it will always fetch or read from cache only depending on `staleTime`.

**Reply by lisherwin** [reply]
Makes sense. Thank you!
